Assets/Plugins/Source/Editor/Windows/EOSEditorWindow.cs

```diff
@@ -121,35 +152,47 @@ protected void SetAutoResize(bool autoResize)
         }
     }
 
+    /// <summary>
+    /// Async function. Base implementation is done, but completes the process of setup by marking
+    /// the window as initialized, so it is important that deriving classes call the base implementation
+    /// at the END of their overriding function.
+    /// </summary>
+    /// <returns></returns>
+    protected virtual async Task AsyncSetup()
+    {
+        // default implementation does nothing.
+        await Task.Run(() => { });
+        Debug.Log("Window has been initialized.");
+    }
+
     /// <summary>
     /// Implement this method to initialize the contents of the editor window. Be sure to always call the base implementation if you override it.
     /// </summary>
     protected virtual void Setup()
     {
-        foreach (var fieldKeyPair in RetainableFields)
+        foreach (var (field, key) in RetainableFields)
```

Comment:
Another thing to check to see if it's supported.

Reply:
What does this Retainable Field do? I couldn't where the attribute was used

Reply:
If you add a `RetainableField` attribute to any field member on a class derived from `EOSEditorWindow`, it's value will be saved and loaded automatically when the window is closed and subsequently opened. 

It's currently not utilized for anything, but the use case that caused it's implementation in another project is that it allowed user preferences to persist for things like whether or not they had opened an expandable area or not. Basically: preferences related to how the window itself operates - nothing that's related directly to the plugin or the samples.